framework/set/resources/rancher2/setProvidersAndUsersTF.go

```diff
@@ -222,18 +162,48 @@ func createGlobalRoleBinding(rootBody *hclwrite.Body, testUser string, userID st
 	globalRoleBindingBlockBody.SetAttributeRaw(userID, standardUser)
 }
 
-// determine if config within configMap contains a custom module
-func containsCustomModule(configMap []map[string]any) bool {
-	customModule := false
-
+// Determines the required providers from the list of configs.
+func getRequiredProviderVersions(configMap []map[string]any) (string, string, string, string, string) {
+	source := ""
```

Comment:
You could probably just do a `var source, rancherProviderVersion, rkeProviderVersion, localProviderVersion, awsProviderVersion string`. So that you don't have to do multiple declarations.